### PR TITLE
Cloud migrations: create route to delete token

### PR DESCRIPTION
**Pull request by PoorlyDefinedBehaviour**
- Creates `DELETE api/cloudmigration/token/:uid` to delete a cloud migration token.
- Always returns 204 unless an error happens.

Closes https://github.com/grafana/grafana-operator-experience-squad/issues/779

### CODE REVIEW

File: pkg/services/gcom/gcom.go [hunk 1]
```diff
@@ -328,3 +336,43 @@ func (client *GcomClient) CreateToken(ctx context.Context, params CreateTokenPar
...
+		return false, fmt.Errorf("sending http request to delete access token: %w", err)
+	}
+	defer func() {
+		if err := response.Body.Close(); err != nil {
+			client.log.Error("closing http response body", "err", err.Error())
+		}
+	}()
+
+	if response.StatusCode == http.StatusNotFound {
+		return false, nil
```
Comment (leandro-deveikis) [inline pkg/services/gcom/gcom.go:369]: This seems a bit confusing to me. I get that we may not want to take the "token not found" as an error, but I feel is not clear that the problem is from the outside. I think I would prefer to remove the bool return value, and handle the "not happy" cases with errors, for instance, add an TokenNotFoundError and return that.

Comment (leandro-deveikis) [reply pkg/services/gcom/gcom.go:369]: Checking the CloudMigrationService, when we make this call, we ignore the bool return value, so there is no need to add this extra logic in my opinion. If we want to ignore the case where the token is not found, I would suggest to do that in the CloudMigrationService so is more clear (if that is the intention)

Comment (PoorlyDefinedBehaviour) [reply pkg/services/gcom/gcom.go]: Done. Returning an error instead of a boolean.

File: pkg/services/gcom/gcom.go [hunk 2]
```diff
@@ -13,14 +14,18 @@ import (
...
+
+var ErrTokenNotFound = errors.New("gcom: token not found")
 
 type Service interface {
 	GetInstanceByID(ctx context.Context, requestID string, instanceID string) (Instance, error)
 	CreateAccessPolicy(ctx context.Context, params CreateAccessPolicyParams, payload CreateAccessPolicyPayload) (AccessPolicy, error)
 	ListAccessPolicies(ctx context.Context, params ListAccessPoliciesParams) ([]AccessPolicy, error)
 	DeleteAccessPolicy(ctx context.Context, params DeleteAccessPolicyParams) (bool, error)
 	CreateToken(ctx context.Context, params CreateTokenParams, payload CreateTokenPayload) (Token, error)
+	// Deletes an auth token.
```
Comment (mmandrus) [inline pkg/services/gcom/gcom.go:27]: nit: comment not needed

File: pkg/services/cloudmigration/cloudmigrationimpl/gcomstub.go [hunk 3]
```diff
@@ -58,3 +58,7 @@ func (client *gcomStub) CreateToken(ctx context.Context, params gcom.CreateToken
 	}
 	return token, nil
 }
+
+func (client *gcomStub) DeleteToken(ctx context.Context, params gcom.DeleteTokenParams) error {
+	return nil
```
Comment (mmandrus) [inline pkg/services/cloudmigration/cloudmigrationimpl/gcomstub.go:63]: After merging https://github.com/grafana/grafana/pull/88176, update this to set `client.token` to nil.

**Review by mmandrus — APPROVED** [review]

**Review by bornohin6 — APPROVED** [review]
LGTM.

Unrelated to this PR, but a suggestion for the UX change: it might be helpful to display a warning message if the token was last used within a predefined time period.

File: pkg/services/gcom/gcom.go (hunk 2 above)
Comment (PoorlyDefinedBehaviour) [reply pkg/services/gcom/gcom.go:27]: Removed

File: pkg/services/cloudmigration/cloudmigrationimpl/gcomstub.go (hunk 3 above)
Comment (PoorlyDefinedBehaviour) [reply pkg/services/cloudmigration/cloudmigrationimpl/gcomstub.go:79]: Done

**Pull request merged**